### PR TITLE
transform.cat should live in the cat namespace.

### PR DESCRIPTION
**Pull request by Mpdreamz**
Similarly to to `ml` cat API's also living in the `cat` namespace.

Clients treat the `cat` namespace differently then other API's (return types, content types). This fixes the exception to this rule.

### CODE REVIEW

File: x-pack/plugin/src/test/resources/rest-api-spec/api/transform.cat_transform.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "transform.cat_transform":{
+  "cat.transform":{
```
Comment (droberts195) [inline x-pack/plugin/src/test/resources/rest-api-spec/api/transform.cat_transform.json]: I think you also need to change the file name to match this

Comment (Mpdreamz) [reply x-pack/plugin/src/test/resources/rest-api-spec/api/transform.cat_transform.json]: :doh: renamed: `x-pack/plugin/**out**/test/resources/rest-api-spec/api/cat.transform.json` but missed this one. thanks @droberts195

**Comment by droberts195** [comment]
I marked this `>non-issue` because by putting it into 7.7 it doesn't need to be in the release notes, not because I don't think it should be done 😃

**Review by droberts195 — APPROVED** [review]
LGTM

Thanks for noticing and fixing the discrepancy!

**Review by benwtrent — APPROVED** [review]
❤️ 

🍻 for @Mpdreamz

**Comment by Mpdreamz** [comment]
Thanks for the reviews @droberts195 @benwtrent 👍

**Pull request merged**